gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java

```diff
@@ -403,41 +453,58 @@ public EntityPlayer teleport(Location to, EntityTeleportEvent.Cause cause) {
         // Reset chunks
         this.connection.resetQueuedChunks();
 
+        Chunk fromChunk = this.chunk();
+
         // Check if we need to change worlds
         if (!to.world().equals(from.world())) {
+            if (this.world != null) {
```

Comment:
this should be from.world() and there is no need for a null check

Reply:
null check removed